openeo/rest/datacube.py

```diff
@@ -441,10 +441,24 @@ def load_stac(
         graph = PGNode("load_stac", arguments=arguments)
         try:
             metadata = metadata_from_stac(url)
+            # TODO: also apply spatial/temporal filters to metadata?
+
             if isinstance(bands, list):
-                # TODO: also apply spatial/temporal filters to metadata?
-                metadata = metadata.filter_bands(band_names=bands)
-        except Exception:
+                if not metadata.has_band_dimension():
+                    metadata = metadata.add_dimension(
+                        name="bands",
+                        type="bands",
+                        label=None,
+                    )
+                missing_bands = [b for b in bands if not metadata.band_dimension.contains_band(b)]
```

Comment:
"missing" is a bit confusing here. Maybe "unknown" fits the situation a bit better